src/ITCC.YandexSpeeckKitClient/Extensions/NetworkingExtensions.cs

```diff
@@ -0,0 +1,104 @@
+                return ms.ToArray();
+            }
+        }
+        public static Task SendMessageAsync(this Stream stream, object messageObject, CancellationToken cancellationToken)
+        {
+            return stream.WriteMessageAsync(messageObject, cancellationToken);
+        }
+        public static async Task<TMessage> GetDeserializedMessageAsync<TMessage>(this Stream stream, CancellationToken cancellationToken)
+            where TMessage : class
+        {
+            var message = await stream.ReadMessageAsync(cancellationToken);
+            return BinaryMessageSerializer.Deserialize<TMessage>(message);
+        }
+
+        private static async Task WriteMessageAsync(this Stream stream, object messageObject, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var serializedMessage = BinaryMessageSerializer.Serialize(messageObject);
+            var sizeBytes = serializedMessage.Length.ToHexBytes();
+
+            await stream.WriteAsync(sizeBytes, 0, sizeBytes.Length, cancellationToken);
+            await stream.WriteAsync(ControlSeq, 0, ControlSeq.Length, cancellationToken);
+            await stream.WriteAsync(serializedMessage, 0, serializedMessage.Length, cancellationToken);
+        }
+        private static async Task<byte[]> ReadMessageAsync(this Stream stream, CancellationToken cancellationToken)
+        {
+            byte[] sizeHexBytes;
+            using (var searchMemoryStream = new MemoryStream())
+            {
+                var searchBuffer = new byte[1];
+                var searchBufferLength = searchBuffer.Length;
+                while (true)
+                {
+                    var received = await stream.ReadAsync(searchBuffer, 0, searchBufferLength, cancellationToken);
+                    if (received == 0)
+                        continue;
+
+                    if (searchBuffer[0] != CarriageReturn)
+                    {
+                        searchMemoryStream.Write(searchBuffer, 0, searchBufferLength);
+                        continue;
+                    }
+
+                    received = await stream.ReadAsync(searchBuffer, 0, searchBufferLength, cancellationToken);
+                    if (received == 0)
+                        throw new EndOfStreamException();
+
+                    if (searchBuffer[0] != LineFeed)
+                    {
+                        searchMemoryStream.Write(searchBuffer, 0, searchBufferLength);
+                        continue;
+                    }
+
+                    break;
+                }
+                sizeHexBytes = searchMemoryStream.ToArray();
+            }
+
+            var messageLength = sizeHexBytes.FromHexBytes();
+
+            var messageBytes = new byte[messageLength];
+            var receivedBytes = await stream.ReadAsync(messageBytes, 0, messageLength, cancellationToken);
+
+            if (receivedBytes < messageLength)
```

Comment:
We cannot simple remove the check - this can cause rare and very sudden fails
The fast path is to catch  `EndOfStreamException` and return error code.
I think we can really catch it for now and implement partial reading later. This approach also makes future APIs backward-compatible: some kind of errors just stop to occur